wafo-server/src/main/java/wafo/wafoserver/link/service/LinkService.java

```diff
@@ -0,0 +1,47 @@
+package wafo.wafoserver.link.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import wafo.wafoserver.account.user.User;
+import wafo.wafoserver.account.user.repository.UserRepository;
+import wafo.wafoserver.folder.Folder;
+import wafo.wafoserver.folder.repository.FolderRepository;
+import wafo.wafoserver.link.Link;
+import wafo.wafoserver.link.controller.dto.LinkCreateRequest;
+import wafo.wafoserver.link.controller.dto.LinkCreateResponse;
+import wafo.wafoserver.link.dto.CreateLinkDto;
+import wafo.wafoserver.link.repository.LinkRepository;
+
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class LinkService {
+    private final LinkRepository linkRepository;
+    private final UserRepository userRepository;
+    private final FolderRepository folderRepository;
+
+    @Transactional
+    public LinkCreateResponse create(LinkCreateRequest linkCreateRequest) {
+
+        User user = userRepository.findById(linkCreateRequest.getUserId()).get();
+        Folder folder = folderRepository.findById(linkCreateRequest.getFolderId()).get();
+
+        Link link = Link.create(
+                new CreateLinkDto(
+                        user,
+                        folder,
+                        linkCreateRequest.getUrl(),
+                        linkCreateRequest.getTitle(),
+                        linkCreateRequest.getFaviconUrl()
+                )
+        );
+
```

Comment:
링크를 트랜잭션 안에서 persist 하는 로직도 추가되어야 할 것 같습니다 !

```suggestion
linkrepository.save(link);
```

아래 테스트가 성공한 이유는 영속화 되지 않은 link를 그대로 DTO로 반환했기 때문에 값은 잘 나와서 통과한 것 같습니다.

Reply:
반영하였습니다 😁